ts/a11y/semantic-enrich.ts

```diff
@@ -177,14 +193,34 @@ export function EnrichedMathItemMixin<N, T, D, B extends Constructor<AbstractMat
           } else {
             mml = this.adjustSelections();
           }
-          this.inputData.enrichedMml = math.math = this.serializeMml(Sre.toEnriched(mml));
+          Sre.setupEngine(document.options.sre);
+          const enriched = Sre.toEnriched(mml);
+          this.generator.setOptions(Object.assign(
+            {}, document.options.sre, {
+              markup: 'ssml',
+              automark: true,
+            }));
+          this.outputData.speech = buildSpeech(
+            this.generator.getSpeech(enriched, enriched),
+            document.options.sre.locale,
+            document.options.sre.rate)[0];
+          this.generator.setOptions({
+            locale: document.options.sre.braille,
+            domain: 'default',
+            style: 'default',
+            modality: 'braille',
+            markup: 'none',
+          });
+          this.outputData.braille = this.generator.getSpeech(enriched, enriched);
+          this.inputData.enrichedMml = math.math = this.serializeMml(enriched);
           math.display = this.display;
           math.compile(document);
           this.root = math.root;
         } catch (err) {
           document.options.enrichError(document, this, err);
         }
       }
+      setAria(this.root, document.options.sre.locale);
```

Comment:
I'm wondering if this call wants to be dependent on the options that cause speech to be generated, since right now this function walks the entire expression tree even if speech hasn't been generated.

Reply:
Speech is now generated by default during enrichment and therefore it does not really make sense to not set the `aria` elements. However, I have now added an `enableSpeech` option to the `EnrichedMathDocument` that allows us to switch of speech generation entirely.